test/azure_blobs_exception_tests.jl

```diff
@@ -35,6 +35,26 @@
             end
         end
 
+        @testset "Insufficient output buffer size multipart" begin
+            input = "1,2,3,4,5,6,7,8,9,1\n" ^ 1_000_000
+            buffer = Vector{UInt8}(undef, 20_000_000)
+            @assert sizeof(input) == 20_000_000
+            @assert sizeof(buffer) == sizeof(input)
+
+            nbytes_written = put_object(codeunits(input), "test100B.csv", config)
+            @test nbytes_written == 20_000_000
+
+            try
+                # Buffer is over multipart threshold but too small for object
+                buffer = Vector{UInt8}(undef, 10_000_000)
```

Comment:
> Buffer is over multipart threshold

what is that threshold in these tests? is it just the default threshold? what happens if we later change that default threshold to > 10_000_000? do we want this test to fail or not?

I think it'd be better to separate out testing "the defaults are what we expect" from testing "the behaviour of `threshold < buffer < object`" i.e. can we pass all these settings to the test `config` directly, rather than relying on the defaults

And perhaps add separate tests that ensure the defaults are as expected, if we think such tests are useful? Maybe we should test the `get_part_size >= get_threshold` if that's important?

Reply:
Good catch, I added an explicit StaticConfig to the test setup, I think that should be good for now.